### PR TITLE
Added section " What is a bear anyway?"

### PR DESCRIPTION
**Pull request by kshitijkg01**
Briefly describes what a bear is and provides links for more information
on bears .

Fixes https://github.com/coala/documentation/issues/284

### CODE REVIEW

**Comment by hemangsk** [comment]
Hey! Your commit short log should also include the name of that file in which the changes are done.   [Refer](http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html#how-to-write-good-commit-messages) 😃

**Comment by arnab1896** [comment]
@kshitijkg01 ; in the meantime i'm adding a WIP tag.

Amend your commit.
If you find difficulty in ammending commit; follow this ----> http://api.coala.io/en/latest/Developers/Git_Basics.html

**Comment by kshitijkg01** [comment]
Oh OK. So can add that now to what I have already committed ?

**Comment by arnab1896** [comment]
yeah sure. You can make all the changes in this branch on your local repo.

After you're done push again to your online forked repo

**Comment by arnab1896** [comment]
@kshitijkg01 
try it on your own.
think for some time...if you still dont get it; let me know.

cheers! :tada:

**Comment by kshitijkg01** [comment]
I am able to change the file itself, but am not able to change the content of the commit.

**Comment by arnab1896** [comment]
go to your local repo.

use this command: git commit -a --amend
this will open up your COMMIT_EDITMSG file that stores your commit.

make the changes.
save that commit like you had done the first time.
then push again:git push myfork --force

this is obviously assuming your fork's name is myfork

**Comment by kshitijkg01** [comment]
I did that and updated the commit. But it is not updating here. So I tried it again but again it did not get updated here.

**Comment by arnab1896** [comment]
Did you push?

**Comment by kshitijkg01** [comment]
I actually use Git Hub Desktop version, in which i can undo my commit, make changes and commit again. So after undoing my commit, I added filename and committed again. But no change was reflected here.

**Comment by arnab1896** [comment]
I'm not well versed with the desktop version.
@hemangsk , you might be able to help?

**Comment by arnab1896** [comment]
@kshitijkg01 , do you have access to a terminal?
Then you might be able to clone your fork and work

**Comment by hemangsk** [comment]
@kshitijkg01 Please use terminal its much easier than GitHub Desktop version, we just gotta do ```git commit --amend``` and then force push as @arnab1896 already mentioned.

**Comment by kshitijkg01** [comment]
Finally !! I realized that. Git Hub Desktop is very difficult to use. I finally had to use terminal.

**Comment by kshitijkg01** [comment]
But I just saw instead of updating the old one it just created a new one. So is that how it is supposed to happen or does it usually update the same commit instead of creating a new one ?

**Comment by hemangsk** [comment]
@kshitijkg01 Hey! You made new commits instead of modifying the existing one. Once we fix a commit after review, we'll want to squash our commits. http://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits

Then what'll happen is, your PR will have a single commit, doing atomic change, without any Merge or Fixup commits  (Right now it has 3 commits). You can find everything in detail in the Git Tutorial. If you get stuck anywhere, you can always ask on gitter as well.

**Comment by kshitijkg01** [comment]
Hey I tried that and in the editor that opens, instead of getting the commits i get noop. I have attached a picture for your reference. Can you please help me out of this ? 
image

**Comment by kshitijkg01** [comment]
Hi I managed it by using the head command "HEAD~3" and it worked. But I still wonder why it did not show all the commits without using that command.

**Comment by tabias** [comment]
Hi, commit looks good.
You're still missing the shortlog ( Filename: Described short )

**Comment by kshitijkg01** [comment]
Filename: help/FAQ.rst. Isn't this enough ?

**Comment by tabias** [comment]
follow the example 
http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html

This should give you a correct example to follow. There is clear difference between the shortlog and the description.

**Comment by arnab1896** [comment]
@kshitijkg01 ;
1.content is fine. :+1: 
2.commit shortlog looks fine too. :+1: 

just two small points i want to tell you.

remember that across all the repositories of of coala there are two kinds of issues. Those that are bugs and those that are not bugs - but are enhancements or additions to the code.
the issue that you have solved isn't a bug.

so-->
1.use the word "Closes #284 " instead of "Fixes #284 ".
2.i think your body's first line has exceeded 72 characters.

rest all seems good.
cheers! :tada:

**Comment by kshitijkg01** [comment]
I thought the maximum limit is 79 characters

**Comment by kshitijkg01** [comment]
I have changed Fixes to Closes and have also made the first line of the body less that 72 characters

**Comment by arnab1896** [comment]
ack ba05f8e

File: conf.py [hunk 1]
```diff
@@ -94,7 +94,7 @@
 #show_authors = False
 
 # The name of the Pygments (syntax highlighting) style to use.
-pygments_style = 'sphinx'
+pygments_style = 'native'
```
Comment (Udayan12167) [inline conf.py]: Unrelated change.

File: Help/FAQ.rst [hunk 2]
```diff
@@ -20,6 +20,23 @@ In short coala does two things:
   everything but the actual analyzer routine. Write just the parameters
   your custom analysis needs and the analysis part, we'll take care of the rest
 
+What is a bear anyway?
```
Comment (Udayan12167) [inline Help/FAQ.rst]: Just "What is a bear?"

File: Help/FAQ.rst [hunk 3]
```diff
@@ -20,6 +20,23 @@ In short coala does two things:
...
+----------------------
+
+A bear is meant to do some analysis on source code. It can check your code for
```
Comment (Udayan12167) [inline Help/FAQ.rst]: `Some analysis` is very vague. Maybe say something like `static analysis...`

File: Help/FAQ.rst [hunk 4]
```diff
@@ -20,6 +20,23 @@ In short coala does two things:
...
+potential problems, calculate metrics and even provide corrections for your
+code. The source code will be provided by coala so the bear doesn’t have to
```
Comment (Udayan12167) [inline Help/FAQ.rst]: `The source code will be provided by coala...` This sentence isn't very clear. Can you maybe reword it?

**Comment by Udayan12167** [comment]
unack ba05f8e

File: conf.py (hunk 1 above)
Comment (kshitijkg01) [inline conf.py]: I had not made thin change, I had updated my branch and this commit came up.

**Comment by kshitijkg01** [comment]
I have made the respective changes

File: conf.py (hunk 1 above)
Comment (tabias) [inline conf.py]: rebasing your commit on the current master should help with this.

Comment (Udayan12167) [inline conf.py]: 👍

**Comment by nemaniarjun** [comment]
@kshitijkg01 I think that your rebase has gone wrong. Your commit got mixed with the last commit in this repo ( one which was by @hemangsk )
Your branch is one commit ahead and one commit behind coala:master, while it should only be one commit ahead.
Squashing guide : http://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits

Fixing this is not straighforward.
This is one of the reasons of why we should never develop on the master branch.

Ask on the gitter channel for more help.

**Comment by kshitijkg01** [comment]
Hi I did as you told. I first forked a branch(master) form this repository(coala/master) and then created one more branch(MyBranch) from master. I then did my changes on that branch and pushed it to master. I then merged MyBranch with the master branch. Now how do i update this pull request and not create a new one

**Comment by jayvdb** [comment]
It isnt possible to change the branch of a PR.  So this PR using 'master' is not usable.  Create a new pull request using your branch ;-)

**Comment by kshitijkg01** [comment]
Oh OK
